File: cpp/06_091.cpp
std::vector<std::vector<int>> kruskalMST(int n,
        std::vector<std::vector<int>>& edges) {
    // Sort edges by weight
    std::sort(edges.begin(), edges.end(),
        [](const std::vector<int>& a, const std::vector<int>& b) {
            return a[2] < b[2];
        });
    
    UnionFind uf(n);
    std::vector<std::vector<int>> mst;
    
    for (const auto& edge : edges) {
        int u = edge[0];
        int v = edge[1];
        int weight = edge[2];
        
        if (uf.unionSets(u, v)) {
            mst.push_back({u, v, weight});
        }
        
        // Early termination when we have n-1 edges
        if (mst.size() == n - 1) {
            break;
        }
    }
    
    return mst;
}

File: data structures and algorithms/cs/06_091.cs
public static int[] FindRedundantConnection(int[][] edges) {
    int n = edges.Length;
    UnionFind uf = new UnionFind(n + 1);

    foreach (int[] edge in edges) {
        int u = edge[0];
        int v = edge[1];
        if (!uf.Union(u, v)) {
            return edge;
        }
    }

    return new int[0];
}